fastapi_router_controller/lib/controller.py

```diff
@@ -88,47 +92,56 @@ def wrapper(cls):
 
     @staticmethod
     def __parse_controller_router(cls):
-        '''
+        """
             Private utility to parse the router controller property and extract the correct functions handlers
-        '''
+        """
         router = getattr(cls, Controller.RC_KEY)
 
+        dependencies = None
+        if hasattr(cls, "dependencies"):
+            dependencies = deepcopy(cls.dependencies)
+            delattr(cls, "dependencies")
+
         for route in router.routes:
             # get the signature of the endpoint function
             signature = inspect.signature(route.endpoint)
             # get the parameters of the endpoint function
             signature_parameters = list(signature.parameters.values())
+            # add class dependencies
+            if dependencies:
+                for depends in dependencies[::-1]:
+                    route.dependencies.insert(0, depends)
```

Comment:
@delijati Does route.dependencies always exists?

Reply:
@delijati I will move it at line 106, just to keep it separated from rewrite sign

Reply:
yes https://github.com/tiangolo/fastapi/blob/master/fastapi/routing.py#L342 and sure